fvm/environment/transaction_info.go

```diff
@@ -0,0 +1,79 @@
+package environment
+
+import (
+	"github.com/onflow/cadence/runtime"
+
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/trace"
+)
+
+type TransactionInfo interface {
+	SigningAccounts() []runtime.Address
+
+	IsServiceAccountAuthorizer() bool
+
+	// Candence's runtime api.
+	GetSigningAccounts() ([]runtime.Address, error)
```

Comment:
better to mention the possible error return would be `OperationNotSupportedError`